### ISSUE TITLE
Bug with using --help for description longer than 128 bytes.

### ISSUE THREAD

**Issue by schuhschuh** [gflags/gflags#59]
Original [issue 48](https://code.google.com/p/gflags/issues/detail?id=48) created by schuhschuh on 2011-08-16T12:12:31.000Z:

<b>What steps will reproduce the problem?</b>
1. Compile version 1.6 using Visual Studio 2008
2. Create a simple main.cpp program as follows:
# include &lt;gflags/gflags.h&gt;

int main(int argc, char\* argv[])
{
    google::ParseCommandLineFlags(&amp;argc, &amp;argv, true);
    return 0;
}
1. Run it with &quot;--help&quot; as the argument.

<b>What is the expected output? What do you see instead?</b>
It will blow up on the first assert in gflags_reporting.cc:DescribeOneFlag(const CommandLineFlagInfo&amp; flag).

This is because the flag.description is truncated in SStringPrintf, since the util.h:InternalStringPrintf() contains a bug.

Basically the space[128] buffer is reused wrongly in the second loop, which tries to reformat the string with a bigger buffer. Instead of using 
  vsnprintf(buf, length, ...)
it is
  vsnprintf(space, length, ...)

Which uses the wrong buffer.

<b>What version of the product are you using? On what operating system?</b>
This bug should be present on most platforms as far as I can tell from the code. Basically, it will currently blow up when running &quot;--help&quot; where the description is longer than 128 bytes.

<b>Please provide any additional information below.</b>

A patch is attached.
